cpp/src/arrow/flight/client_header_auth_middleware.h

```diff
@@ -0,0 +1,78 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
This should probably be an _internal.h header since I don't think any of this is intended to be directly used outside of the implementation here.